### PR TITLE
nep1: prefer `a..b` to `a .. b` except if `b` has an operator (eg: `a .. -1`)

### PR DESCRIPTION
**Pull request by timotheecour**
refs https://github.com/nim-lang/Nim/pull/16981#discussion_r573371075

nep1 concentrates debate in 1 place instead of debating it in each PR

### CODE REVIEW

File: doc/nep1.rst [hunk 1]
```diff
@@ -280,3 +280,6 @@ Conventions for multi-line statements and expressions
   .. code-block:: nim
     startProcess(nimExecutable, currentDirectory, compilerArguments
                  environment, processOptions)
+
+- Use `a..b` instead of `a .. b`, except when `b` contains an operator, e.g `a .. -3`
```
Comment (juancarlospaco) [inline doc/nep1.rst]: No eggs https://github.com/nim-lang/Nim/pull/15906#issuecomment-724879032

Please use `example`, easy for non-English native people, explicit, not ambiguous.

File: doc/nep1.rst [hunk 2]
```diff
@@ -280,3 +280,6 @@ Conventions for multi-line statements and expressions
...
+  Likewise with `a..<b`.
```
Comment (juancarlospaco) [inline doc/nep1.rst]: `Likewise` is capitalized, but theres no `.` nor `;` on the previous line, feels like part of the same phrase (?)

**Comment by juancarlospaco** [comment]
I think https://nim-lang.github.io/Nim/contributing.html#best-practices should be cleaned and moved here,
feels like duplication of efforts, DRY.

I like the change.

File: doc/nep1.rst (hunk 1 above)
Comment (timotheecour) [reply doc/nep1.rst]: done

File: doc/nep1.rst (hunk 2 above)
Comment (timotheecour) [reply doc/nep1.rst]: done

**Comment by timotheecour** [comment]
> should be cleaned and moved here

I think the distinction is:
* nep1 is for style (ie enforcing conventions etc)
* https://nim-lang.github.io/Nim/contributing.html#best-practices is for semantics (where rationale can be explained by logic rather than convention)

**Comment by juancarlospaco** [comment]
Okay.

File: doc/nep1.rst [hunk 3]
```diff
@@ -280,3 +280,6 @@ Conventions for multi-line statements and expressions
   .. code-block:: nim
     startProcess(nimExecutable, currentDirectory, compilerArguments
                  environment, processOptions)
+
+- Use `a..b` instead of `a .. b`, except when `b` contains an operator, for example `a .. -3`.
+  Likewise with `a..<b`.
```
Comment (metagn) [inline doc/nep1.rst]: What about `a..^b`? Would the correct style be `a .. ^b`?

Comment (timotheecour) [reply doc/nep1.rst]: `a..^b` is actually `a ..^ b`, not `a .. ^b`; I've changed the wording

**Pull request merged**